File: C/Algorithms/1518.c
int numWaterBottles(int numBottles, int numExchange) {
  int drinks = numBottles;
  while (1) {
    int d = numBottles / numExchange, r = numBottles % numExchange;
    if (d == 0)
      break;
    drinks += d;
    numBottles = d + r;
  }
  return drinks;
}

File: Rust/Algorithms/1518.rs
impl Solution {
    pub fn num_water_bottles(
        mut num_bottles: i32,
        num_exchange: i32,
    ) -> i32 {
        let mut drinks = num_bottles;
        loop {
            let (d, r) =
                (num_bottles / num_exchange, num_bottles % num_exchange);
            if d == 0 {
                break;
            }
            drinks += d;
            num_bottles = d + r;
        }
        drinks
    }
}
